src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -377,26 +455,33 @@ InsertValidationResponse insertBufferedRecords(StreamingBuffer streamingBufferTo
 
     // intermediate buffer can be empty here if time interval reached but kafka produced no records.
     if (streamingBufferToInsert.isEmpty()) {
-      LOGGER.debug("No Rows Buffered, returning;");
+      LOGGER.debug("No Rows Buffered for channel:{}, returning", this.getChannelName());
       this.previousFlushTimeStampMs = System.currentTimeMillis();
       return null;
     }
+    InsertValidationResponse response = null;
+    try {
+      response = insertRowsWithFallback(streamingBufferToInsert);
+      // Updates the flush time (last time we called insertRows API)
+      this.previousFlushTimeStampMs = System.currentTimeMillis();
 
-    InsertValidationResponse response = insertRowsWithFallback(streamingBufferToInsert);
-    // Updates the flush time (last time we called insertRows API)
-    this.previousFlushTimeStampMs = System.currentTimeMillis();
-
-    LOGGER.info(
-        "[INSERT_ROWS] Successfully called insertRows for channel:{}, noOfRecords:{},"
-            + " startOffset:{}, endOffset:{}, hasErrors:{}",
-        this.getChannelName(),
-        streamingBufferToInsert.getNumOfRecords(),
-        streamingBufferToInsert.getFirstOffset(),
-        streamingBufferToInsert.getLastOffset(),
-        response.hasErrors());
-    if (response.hasErrors()) {
-      handleInsertRowsFailures(
-          response.getInsertErrors(), streamingBufferToInsert.getSinkRecords());
+      LOGGER.info(
+          "Successfully called insertRows for channel:{}, buffer:{}, insertResponseHasErrors:{}",
+          this.getChannelName(),
+          streamingBufferToInsert,
+          response.hasErrors());
+      if (response.hasErrors()) {
+        handleInsertRowsFailures(
+            response.getInsertErrors(), streamingBufferToInsert.getSinkRecords());
+      }
+      return response;
+    } catch (TopicPartitionChannelInsertionException ex) {
+      // Suppressing the exception because other channels might still continue to ingest
```

Comment:
Eh? I don't think we should be suppressing errors here? Can you explain why we'd want to swallow exceptions?

Reply:
Oh I see, so this is partially related to how KC handles failures?

Reply:
Correct, more of how this doesnt qualify for `ConnectException` which can cause the `SinkTask` to fail.